agent/src/capability/configstore.cpp

```diff
@@ -0,0 +1,174 @@
+
+        std::vector<Capability> configCaps = capsFromConfig(configFilePath);
+
+        caps.insert(caps.end(), configCaps.begin(), configCaps.end());
+    }
+
+    return caps;
+}
+*/
+
+std::vector<GatewayConfiguration> softwarecontainer::getGatewaysByCap(std::string capID,
+								      const std::string &filePath) {
+    std::vector<GatewayConfiguration> gateways;
+
+    JSonElement root = JSonElement::fromFile(filePath);
+
+    if (root.isValid()) {
+
+        auto arrayElement = root.getField("capabilities");
+
+        if (arrayElement.isArray()) {
+            std::vector<JSonElement> children;
+            arrayElement.readChildren(children);
+            for (auto& child : children) {
+
+                std::string capName;
+                child.read("name", capName);
+
+		// TODO (thenor): Only push gateways for capID
+		if (capName == capID) {
+		    auto gatewaysElement = child.getField("gateways");
+		    if (arrayElement.isArray()) {
+
+		        std::string strError;
+			if (!parseGatewayData(gatewaysElement, gateways, &strError)) {
+			  log_debug() << "Error in " << pathString << ": " << strError;
+			  gateways.clear();
+			}
+			else {
+			  GatewayConfiguration gw = GatewayConfiguration(gatewayName, gwConf);
+			  gateways.push_back(gw);
+			}
+		    }
+		} else {
+		  log_debug() << "Capability " << capName << " does not match " << capID;
+		}
+            }
+        } else {
+            log_debug() << "Error in " << pathString << " : capabilities must be an array";
+        }
+    } else {
+        log_error() << "Error in JSON file : " << pathString;
+    }
+
+    return gateways;
+}
+
+
+bool softwarecontainer::parseGatewayData(const JSonElement &gatewayData,
+					 std::vector<GatewayConfiguration> &gateways,
+					 std::string *error)
+{
+    bool success = false;
+
+    std::vector<JSonElement> gatewayElements;
+    gatewayData.readChildren(gatewayElements);
+
+    for(auto& gatewayElement: gatewayElements) {
+
+        std::string gwId;
+        gatewayElement.read("id", gwId);
+
+        Gateway gw = Gateway(gwId, gatewayElement.getField("config").dump());
+        gateways.push_back(gw);
+
+        success = true;
+    }
+
+    return success;
+}
+
+std::vector<std::string> softwarecontainer::configFileList(const std::string &configPath)
+{
+    std::vector<std::string> configFiles;
+
+    DIR *dirFile = opendir(configPath.c_str());
+    if (dirFile == nullptr) {
+        log_debug() << "Could not open directory: " << configPath << "\n";
+    } else {
+        struct dirent *hFile;
+        while ((hFile = readdir(dirFile)) != nullptr) {
+            std::string filename(hFile->d_name);
+
+            if (pelagicore::has_suffix(filename, ".json"))
+                configFiles.push_back(filename);
+        }
+        closedir(dirFile);
+    }
+
+    return configFiles;
+}
+
+bool softwarecontainer::hasEnding(std::string const &fullString, std::string const &ending)
```

Comment:
These  functions are declared in the softwarecontainer::ConfigStore in the header file, but here they are in softwarecontaner:: only?